jiant/tasks/lib/templates/mlm_premasked.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import torch
+from dataclasses import dataclass
+from typing import List
+
+from jiant.tasks.lib.templates import mlm as mlm_template
+from jiant.tasks.core import (
+    BaseExample,
+    BaseTokenizedExample,
+    BaseDataRow,
+    BatchMixin,
+)
+from jiant.tasks.utils import ExclusiveSpan, truncate_sequences
+from jiant.tasks.lib.templates.shared import (
+    construct_single_input_tokens_and_segment_ids,
+    create_input_set_from_tokens_and_segments,
+    pad_single_with_feat_spec,
+)
+
+
+@dataclass
+class Example(BaseExample):
+    guid: str
+    text: str
+    # Spans over char indices
+    masked_spans: List[ExclusiveSpan]
+
+    def tokenize(self, tokenizer):
+        # masked_tokens will be regular tokens except with tokenizer.mask_token for masked spans
+        # label_tokens will be tokenizer.pad_token except with the regular tokens for masked spans
+        masked_tokens = []
+        label_tokens = []
+        curr = 0
+        for start, end in self.masked_spans:
+            # Handle text before next mask
+            tokenized_text = tokenizer.tokenize(self.text[curr:start])
+            masked_tokens += tokenized_text
+            label_tokens += [tokenizer.pad_token] * len(tokenized_text)
+
+            # Handle mask
+            tokenized_masked_text = tokenizer.tokenize(self.text[start:end])
+            masked_tokens += [tokenizer.mask_token] * len(tokenized_masked_text)
+            label_tokens += tokenized_masked_text
+            curr = end
+        if curr < len(self.text):
+            tokenized_text = tokenizer.tokenize(self.text[curr:])
+            masked_tokens += tokenized_text
+            label_tokens += [tokenizer.pad_token] * len(tokenized_text)
+
+        return TokenizedExample(
+            guid=self.guid, masked_tokens=masked_tokens, label_tokens=label_tokens,
+        )
+
+
+@dataclass
+class TokenizedExample(BaseTokenizedExample):
+    guid: str
+    masked_tokens: List
+    label_tokens: List
+
+    def featurize(self, tokenizer, feat_spec):
```

Comment:
Discussed offline - potential for a test here to ensure featurization is correct

Reply:
Added